request/src/main/java/com/vimeo/networking2/internal/VimeoApiClientImpl.kt

```diff
@@ -0,0 +1,835 @@
+        }
+        if (albumPrivacy.password != null) {
+            body[ApiConstants.Parameters.PARAMETER_ALBUM_PASSWORD] = requireNotNull(albumPrivacy.password)
+        }
+        val uri = album.uri ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        return vimeoService.editAlbum(authHeader, uri, body).enqueue(callback)
+    }
+
+    override fun deleteAlbum(album: Album, callback: VimeoCallback<Unit>): VimeoRequest {
+        val uri = album.uri ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        return vimeoService.delete(authHeader, uri, emptyMap()).enqueue(callback)
+    }
+
+    override fun addToAlbum(album: Album, video: Video, callback: VimeoCallback<Unit>): VimeoRequest {
+        val albumUri = album.uri ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        val videoUri = video.uri ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        return vimeoService.addToAlbum(authHeader, albumUri, videoUri).enqueue(callback)
+    }
+
+    override fun removeFromAlbum(album: Album, video: Video, callback: VimeoCallback<Unit>): VimeoRequest {
+        val albumUri = album.uri ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        val videoUri = video.uri ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        return vimeoService.removeFromAlbum(authHeader, albumUri, videoUri).enqueue(callback)
+    }
+
+    override fun modifyVideosInAlbum(
+        album: Album,
+        modificationSpecs: ModifyVideosInAlbumSpecs,
+        callback: VimeoCallback<VideoList>
+    ): VimeoRequest {
+        val uri = album.uri ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        return vimeoService.modifyVideosInAlbum(authHeader, uri, modificationSpecs).enqueue(callback)
+    }
+
+    override fun modifyVideoInAlbums(
+        video: Video,
+        modificationSpecs: ModifyVideoInAlbumsSpecs,
+        callback: VimeoCallback<AlbumList>
+    ): VimeoRequest {
+        val uri = video.uri ?: return localVimeoCallAdapter.enqueueEmptyUri(callback)
+        return vimeoService.modifyVideoInAlbums(authHeader, uri, modificationSpecs).enqueue(callback)
+    }
+
+    @Suppress("ComplexMethod")
+    override fun editVideo(
+        uri: String,
+        title: String?,
+        description: String?,
+        password: String?,
+        commentPrivacyType: CommentPrivacyType?,
+        allowDownload: Boolean?,
+        allowAddToCollections: Boolean?,
+        embedPrivacyType: EmbedPrivacyType?,
+        viewPrivacyType: ViewPrivacyType?,
+        parameters: Map<String, Any>?,
+        callback: VimeoCallback<Video>
+    ): VimeoRequest {
+        val body = parameters.intoMutableMap()
+        if (title != null) {
+            body[ApiConstants.Parameters.PARAMETER_VIDEO_NAME] = title
+        }
+        if (description != null) {
+            body[ApiConstants.Parameters.PARAMETER_VIDEO_DESCRIPTION] = description
+        }
+        if (password != null) {
```

Comment:
Follow up to my other question regarding validation, this is a good specific example. The previous code was doing a client side check to make sure that password was not null when the privacy type was set to password. Do we know what the client/server consequences are of letting errors like this one pass through?

Reply:
In the case of passing a null password with a privacy set to password, the API response as expected with an error indicating that the password is missing. Response below:
```
{
  "invalid_parameters": [
    {
      "field": "password",
      "error_code": 2233,
      "error": "Please enter a password for this video.",
      "developer_message": "User is attempting to set video privacy to `password` without providing a password."
    }
  ],
  "error": "You have provided an invalid parameter. Please contact developer of this application.",
  "link": null,
  "developer_message": "The parameters passed to this API endpoint didn't pass Vimeo's validation. Please check the invalid_parameters list for more information.",
  "error_code": 2204
}
```